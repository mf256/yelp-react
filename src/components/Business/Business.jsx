import React from 'react';
import { PropTypes } from 'prop-types';

import './Business.css';

class Business extends React.PureComponent {
  render() {
    const {
      business: {
        imageSrc, name, address, city, zipCode, category, rating, reviewCount,
      },
    } = this.props;

    return (
      <div className="Business">
        <div className="image-container">
          <img src={imageSrc} alt={name} />
        </div>
        <h2>{name}</h2>
        <div className="Business-information">
          <div className="Business-address">
            <p>{address}</p>
            <p>{city}</p>
            <p>
              {city}
              {' '}
              {zipCode}
            </p>
          </div>
          <div className="Business-reviews">
            <h3>{category}</h3>
            <h3 className="rating">
              {rating}
              {' '}
              stars
            </h3>
            <p>
              {reviewCount}
              {' '}
              reviews
            </p>
          </div>
        </div>
      </div>
    );
  }
}

export const businessType = PropTypes.shape({
  imageSrc: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  address: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  zipCode: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
  reviewCount: PropTypes.number.isRequired,
});

Business.propTypes = {
  business: businessType.isRequired,
};
export default Business;
