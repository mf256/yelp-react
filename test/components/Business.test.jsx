import React from 'react';
import { shallow } from 'enzyme';
import Business from '../../src/components/Business/Business';
import { mockBusiness } from '../mock/mockData';

describe('Business', () => {
  it('should match snapshot', () => {
    const component = shallow(<Business business={mockBusiness} />);

    expect(component).toMatchSnapshot();
  });
});
