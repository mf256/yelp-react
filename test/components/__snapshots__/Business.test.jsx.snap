// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Business should match snapshot 1`] = `
<styled.div
  className="Business"
>
  <styled.figure>
    <styled.img
      alt="Test Name"
      src="https://test.com/test.jpg"
    />
  </styled.figure>
  <styled.h2>
    Test Name
  </styled.h2>
  <styled.div>
    <styled.div>
      <styled.p>
        1 Test Address
      </styled.p>
      <styled.p>
        Test City
      </styled.p>
      <styled.p>
        Test City
         
        10102
      </styled.p>
    </styled.div>
    <styled.div>
      <styled.p>
        Test Category
      </styled.p>
      <styled.p>
        4.5
          stars
      </styled.p>
      <styled.p>
        90
          reviews
      </styled.p>
    </styled.div>
  </styled.div>
</styled.div>
`;
